File: src/main/java/com/example/dpop/id_fsc/FscCodeRepository.java
package com.example.dpop.id_fsc;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface FscCodeRepository extends JpaRepository<FscCode, Long> {

    Optional<FscCode> findByPersonId(Long personId);

    Optional<FscCode> findByPersonIdAndCode(Long personId, String code);
}

File: src/main/kotlin/com/example/dpop/id_fsc/internal/FscCodeRepository.kt
package com.example.dpop.id_fsc.internal

import org.springframework.data.jpa.repository.JpaRepository

interface FscCodeRepository : JpaRepository<FscCode, Long> {
    fun findByPersonId(personId: Long): FscCode?
    fun findByPersonIdAndCodeHash(personId: Long, codeHash: String): FscCode?
}
